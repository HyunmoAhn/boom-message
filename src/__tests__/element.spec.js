import {
  createButton, createMessage, createSelect,
} from '../element';

describe('test element.js', () => {
  it('should match to snapshot about createButton', () => {
    const wrapper = createButton('mock');

    expect(wrapper).toMatchSnapshot();
  });

  it('should match to snapshot about createSelect', () => {
    const wrapper = createSelect([
      {
        value: 3,
        text: '3초',
      },
      {
        value: 5,
        text: '5초',
      },
      {
        value: '2-times',
        text: '2배',
      },
      {
        value: '3-times',
        text: '3배',
      },
    ]);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match to snapshot about createMessage', () => {
    const wrapper = createMessage('mock-id', 'mock-msg', 10);

    expect(wrapper).toMatchSnapshot();
  });

  describe('test createMessage onClick', () => {
    const mockStore = {
      deleteMessage: jest.fn(),
      updateTime: jest.fn(),
    };

    beforeEach(() => {
      mockStore.deleteMessage.mockClear();
      mockStore.updateTime.mockClear();
    });

    it('should test plusBtn click', () => {
      const wrapper = createMessage('mock-id', 'mock-msg', 10, mockStore);

      const plusSelect = wrapper.querySelector('.time-plus-select');
      const plusBtn = wrapper.querySelector('.time-plus-btn');
      plusSelect.value = 5;

      plusBtn.onclick();

      expect(mockStore.updateTime).toHaveBeenCalledTimes(1);
      expect(mockStore.updateTime).toHaveBeenCalledWith('mock-id', "5");
    });

    it('should test minusBtn click', () => {
      const wrapper = createMessage('mock-id', 'mock-msg', 10, mockStore);

      const minusSelect = wrapper.querySelector('.time-minus-select');
      const minusBtn = wrapper.querySelector('.time-minus-btn');
      minusSelect.value = -5;

      minusBtn.onclick();

      expect(mockStore.updateTime).toHaveBeenCalledTimes(1);
      expect(mockStore.updateTime).toHaveBeenCalledWith('mock-id', "-5");
    });

    it('should test deleteBtn click', () => {
      const wrapper = createMessage('mock-id', 'mock-msg', 10, mockStore);

      const deleteBtn = wrapper.querySelector('.message-delete-btn');

      deleteBtn.onclick();

      expect(mockStore.deleteMessage).toHaveBeenCalledTimes(1);
      expect(mockStore.deleteMessage).toHaveBeenCalledWith('mock-id');
    });
  });
});
