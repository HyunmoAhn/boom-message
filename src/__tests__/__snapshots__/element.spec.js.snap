// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test element.js should match to snapshot about createButton 1`] = `
<button>
  mock
</button>
`;

exports[`test element.js should match to snapshot about createMessage 1`] = `
<div
  class="message"
>
  <p>
    mock-msg
     남은시간: 
    <span
      class="message-time"
    >
      10
    </span>
  </p>
  <p
    class="message-controller"
  >
    <select
      class="time-plus-select"
    >
      <option
        value="3"
      >
        3초
      </option>
      <option
        value="5"
      >
        5초
      </option>
      <option
        value="2-times"
      >
        2배
      </option>
      <option
        value="3-times"
      >
        3배
      </option>
    </select>
    <button
      class="time-plus-btn"
    >
      시간 추가
    </button>
    <select
      class="time-minus-select"
    >
      <option
        value="-3"
      >
        -3초
      </option>
      <option
        value="-5"
      >
        -5초
      </option>
    </select>
    <button
      class="time-minus-btn"
    >
      시간 감소
    </button>
    <button
      class="message-delete-btn"
    >
      삭제
    </button>
  </p>
</div>
`;

exports[`test element.js should match to snapshot about createSelect 1`] = `
<select>
  <option
    value="3"
  >
    3초
  </option>
  <option
    value="5"
  >
    5초
  </option>
  <option
    value="2-times"
  >
    2배
  </option>
  <option
    value="3-times"
  >
    3배
  </option>
</select>
`;
